be/src/runtime/thread_mem_tracker_mgr.h

```diff
@@ -175,39 +170,70 @@ class ThreadMemTrackerMgr {
     ConsumeErrCallBackInfo _consume_err_cb;
 };
 
+inline void ThreadMemTrackerMgr::init() {
+    _tracker_id = 0;
+    _mem_trackers.clear();
+    _mem_trackers[0] = MemTracker::get_process_tracker();
+    _untracked_mems.clear();
+    _untracked_mems[0] = 0;
+    _mem_tracker_labels.clear();
+    _mem_tracker_labels[0] = MemTracker::get_process_tracker()->label();
+}
+
+inline void ThreadMemTrackerMgr::init_bthread() {
+    init();
+    _mem_trackers[1] = MemTracker::get_brpc_server_tracker();
+    _untracked_mems[1] = 0;
+    _mem_tracker_labels[1] = MemTracker::get_brpc_server_tracker()->label();
+    _tracker_id = 1;
+}
+
+inline void ThreadMemTrackerMgr::clear_untracked_mems() {
+    for (const auto& untracked_mem : _untracked_mems) {
+        if (untracked_mem.second != 0) {
+            DCHECK(_mem_trackers[untracked_mem.first]) << print_debug_string();
+            _mem_trackers[untracked_mem.first]->consume(untracked_mem.second);
+        }
+    }
+    mem_tracker()->consume(_untracked_mem);
+    _untracked_mem = 0;
+}
+
 template <bool Existed>
 inline int64_t ThreadMemTrackerMgr::update_tracker(const std::shared_ptr<MemTracker>& mem_tracker) {
-    DCHECK(mem_tracker);
+    DCHECK(mem_tracker) << print_debug_string();
     _temp_tracker_id = mem_tracker->id();
     if (_temp_tracker_id == _tracker_id) {
         return _tracker_id;
     }
     if (Existed) {
-        DCHECK(_mem_trackers.find(_temp_tracker_id) != _mem_trackers.end());
+        DCHECK(_mem_trackers.find(_temp_tracker_id) != _mem_trackers.end()) << print_debug_string();
     } else {
         if (_mem_trackers.find(_temp_tracker_id) == _mem_trackers.end()) {
```

Comment:
The purpose of map find here:
If the tracker has already been added, avoid `_untracked_mems[_temp_tracker_id] = 0;` causing the memory track to be lost.
![image](https://user-images.githubusercontent.com/13197424/164979607-2990184a-8d01-4f67-b4ac-0dfd7c808522.png)

(So in order to avoid this map find, you should add tracker as early as possible, and then call update_tracker<Existed>)